Cargo.toml

```diff
@@ -45,15 +45,42 @@ ptr_as_ptr = "warn"
 ptr_cast_constness = "warn"
 ref_as_ptr = "warn"
 
+std_instead_of_core = "warn"
+std_instead_of_alloc = "warn"
+alloc_instead_of_core = "warn"
```

Comment:
These lints direct Bevy contributors to use `core` where possible, then `alloc`, and then finally `std`. The status quo is to just import everything from `std`. Enabling these lints makes investigation into a `no_std` Bevy feasible. Additionally, items in `alloc` and `std` but not in `core` can indicate blocking or allocating types, which could impact performance.